### PR TITLE
Feature/delete additional info

### PR DESCRIPTION
**Pull request by StrongCurrent**
Now you can finally also delete additional information to 🥳.

### CODE REVIEW

File: db/models/spots.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import mongoose from "mongoose";
 const { Schema } = mongoose;
 
 const AdditionalInformationSchema = new Schema({
+  _id: { type: Schema.Types.ObjectId },
```
Comment (Kheirah) [inline db/models/spots.js:6]: I can see you're explicitly managing the `_id` key and manually saving/updating/deleting to MongoDB (in your endpoints), however MongoDB manages its `ids` internally. You don't need to do it yourself.

Is there a reason for doing everyting manually?

File: components/SpotInfo/index.js [hunk 2]
```diff
@@ -7,49 +7,35 @@ import {
...
 import Error from "../Error";
+import { PiTrash, PiPencilLight } from "react-icons/pi";
 
 export default function SpotInfo({ spotId }) {
-  const {
-    data: spot,
-    error,
-    isValidating,
-  } = useSWR(`/api/spots/${spotId}`);
+  const { data: spot, error, isValidating } = useSWR(`/api/spots/${spotId}`);
```
Comment (Kheirah) [inline components/SpotInfo/index.js:28]: If you want to revalidate your cache, destructure `mutate` right here. And then use it below as `mutate()` only. Remove the import of `mutate` at the top level of this file.

File: components/SpotInfo/index.js [hunk 3]
```diff
@@ -61,10 +47,30 @@ export default function SpotInfo({ spotId }) {
...
+  const handleDeleteEntry = async (infoId) => {
+    const response = await fetch(`/api/spots/${spotId}/informations/${infoId}`, {
+      method: "Delete",
+    });
+
+    if (response.ok) {
+      const data = await response.json();
+
+      if (data && data.success) {
+        mutate(`/api/spots/${spotId}`, undefined, true);
```
Comment (Kheirah) [inline components/SpotInfo/index.js:62]:
```suggestion
        mutate();
```
=)

File: components/SpotInfo/index.js [hunk 4]
```diff
@@ -61,10 +47,30 @@ export default function SpotInfo({ spotId }) {
     if (response.ok) {
       setNewInfo("");
       mutate(`/api/spots/${spotId}`);
```
Comment (Kheirah) [inline components/SpotInfo/index.js:49]:
```suggestion
      mutate();
```

**Review by Kheirah — CHANGES_REQUESTED** [review]
Nice work!

Check the use of `mutate` below.
Also, why are you managing the communication to MongoDB by manually creating ids? Nothing wrong with that, I'm just wondering :)

File: db/models/spots.js (hunk 1 above)
Comment (StrongCurrent) [reply db/models/spots.js:6]: Good question 😅 the reason is that i had serious problems getting it work yesterday. So i tried to define the key manually. now that it works, it works without the manual key. thanks for the hint

**Comment by StrongCurrent** [comment]
Thanks for the feedback i have implemented the suggested changes.

File: components/SpotInfo/index.js [hunk 5]
```diff
@@ -46,22 +50,25 @@ export default function SpotInfo({ spotId }) {
...
   };
 
   const handleDeleteEntry = async (infoId) => {
-    const response = await fetch(`/api/spots/${spotId}/informations/${infoId}`, {
-      method: "Delete",
-    });
+    const response = await fetch(
+      `/api/spots/${spotId}/informations/${infoId}`,
+      {
+        method: "Delete",
```
Comment (moonwave99) [inline components/SpotInfo/index.js:61]: `DELETE` in all caps ^^

**Review by moonwave99 — CHANGES_REQUESTED** [review]

**Review by moonwave99 — APPROVED** [review]

**Pull request merged**